Tests/js/views/assets/ez-dashboardblockbasedview-tests.js

```diff
@@ -0,0 +1,222 @@
+/*
+ * Copyright (C) eZ Systems AS. All rights reserved.
+ * For full copyright and license information view LICENSE file distributed with this source code.
+ */
+YUI.add('ez-dashboardblockbasedview-tests', function (Y) {
+    var renderTest,
+        eventsTest,
+        publicMethodsTest,
+        VIEW_CONFIG = {container: '.container'},
+        SELECTOR_CONTENT = '.dashboard-content',
+        View = Y.eZ.DashboardBlockBasedView;
+
+    Y.eZ.DashboardBlockBaseView = Y.Base.create('dashboardBlockBaseView', Y.View, [], {}, {
+        ATTRS: {
+           priority: {},
+            identifier: {}
+        }
+    });
+
+    renderTest = new Y.Test.Case({
+        name: 'eZ Dashboard Block Based View render test',
+
+        setUp: function () {
+            this.view = new View(VIEW_CONFIG);
+        },
+
+        tearDown: function () {
+            this.view.destroy();
+        },
+
+        'Should render view correctly without blocks inside': function () {
+            var templateCalled = false,
+                origTpl = this.view.template;
+
+            this.view.template = function () {
+                templateCalled = true;
+
+                return origTpl.apply(this, arguments);
+            };
+
+            this.view.render();
+
+            Y.Assert.isTrue(templateCalled, 'The template should have been used to render view');
+        },
+
+        'Should render view correctly with blocks inside': function () {
+            var templateCalled = false,
+                origTpl = this.view.template;
+
+            this.view.template = function () {
+                templateCalled = true;
+
+                return origTpl.apply(this, arguments);
+            };
+
+            this.view._set('blocks', [
```

Comment:
and ? addBlock should be usable before rendering. Actually, this is connected to this thread above https://github.com/ezsystems/PlatformUIBundle/pull/584#discussion_r62852782 and as written there using it before rendering is the main use case for now.

Remove the `blockChange` event handler, and you'll see, this will become easier. (that's exactly the point of all my comments on removing not necessary code...).